ui/src/app/views/app/component.ts

```diff
@@ -37,37 +38,43 @@ export class AppComponent implements OnInit {
   ) {}
 
   ngOnInit(): void {
-    this.overriddenUrl = localStorage.getItem(overriddenUrlKey);
-    this.overriddenPublicUrl = localStorage.getItem(overriddenPublicUrlKey);
-    window['setAllOfUsApiUrl'] = (url: string) => {
-      if (url) {
-        if (!url.match(/^https?:[/][/][a-z0-9.:-]+$/)) {
-          throw new Error('URL should be of the form "http[s]://host.example.com[:port]"');
+    try {
+      this.overriddenUrl = localStorage.getItem(overriddenUrlKey);
+      this.overriddenPublicUrl = localStorage.getItem(overriddenPublicUrlKey);
+      window['setAllOfUsApiUrl'] = (url: string) => {
+        if (url) {
+          if (!url.match(/^https?:[/][/][a-z0-9.:-]+$/)) {
+            throw new Error('URL should be of the form "http[s]://host.example.com[:port]"');
+          }
+          this.overriddenUrl = url;
+          localStorage.setItem(overriddenUrlKey, url);
+        } else {
+          this.overriddenUrl = null;
+          localStorage.removeItem(overriddenUrlKey);
         }
-        this.overriddenUrl = url;
-        localStorage.setItem(overriddenUrlKey, url);
-      } else {
-        this.overriddenUrl = null;
-        localStorage.removeItem(overriddenUrlKey);
-      }
-      window.location.reload();
-    };
-    window['setPublicApiUrl'] = (url: string) => {
-      if (url) {
-        if (!url.match(/^https?:[/][/][a-z0-9.:-]+$/)) {
-          throw new Error('URL should be of the form "http[s]://host.example.com[:port]"');
+        window.location.reload();
+      };
+      window['setPublicApiUrl'] = (url: string) => {
+        if (url) {
+          if (!url.match(/^https?:[/][/][a-z0-9.:-]+$/)) {
+            throw new Error('URL should be of the form "http[s]://host.example.com[:port]"');
+          }
+          this.overriddenPublicUrl = url;
+          localStorage.setItem(overriddenPublicUrlKey, url);
+        } else {
+          this.overriddenPublicUrl = null;
+          localStorage.removeItem(overriddenPublicUrlKey);
         }
-        this.overriddenPublicUrl = url;
-        localStorage.setItem(overriddenPublicUrlKey, url);
-      } else {
-        this.overriddenPublicUrl = null;
-        localStorage.removeItem(overriddenPublicUrlKey);
-      }
-      window.location.reload();
-    };
-    console.log('To override the API URLs, try:\n' +
-      'setAllOfUsApiUrl(\'https://host.example.com:1234\')\n' +
-      'setPublicApiUrl(\'https://host.example.com:5678\')');
+        window.location.reload();
+      };
+      console.log('To override the API URLs, try:\n' +
+        'setAllOfUsApiUrl(\'https://host.example.com:1234\')\n' +
+        'setPublicApiUrl(\'https://host.example.com:5678\')');
+    } catch (err) {
+      this.localStorageDisabled = true;
```

Comment:
browser local storage is not the same thing as cookies. Disabling local storage may also break the site, but if so the error messaging should be different.

Reply:
In my testing, disabling cookies has the effect of disabling local storage. In effect, trying to use local storage is what breaks in the app when cookies are disabled in the browser.

Reply:
Got it - they may be grouped under the same browser setting in Chrome. I'm not sure this will be generally true in all browsers, perhaps just modifying the messaging to instead say "local storage and cookies" would be more accurate.

Alternatively, these lines of code are for debug purposes only and are not necessary for the application to function. Is this the only thing that breaks in the app (I'm guessing not)? We could also just log an error here instead or remove this functionality entirely if no one's using it (I think having this debug backdoor in the production build is fairly questionable anyways).

Reply:
I did not test all browsers - just firefox and chrome. Both have the same effect on local storage when cookies are disabled, i.e. local storage is also disabled. 

There are only places where *we* refer to local storage, but it is used in some libraries. I think the main one we care about is angular-2-local-storage (not sure how that's pulled in, or if it's just part of angular). I'm not opposed to removing that functionality altogether ... but it does seem like a separate ticket to me. 

Let me do a little bit of rework and see if I can find a more angular-idiomatic way of getting the right info. That way we can keep the cookie-specific user-messaging simple.

Reply:
FWIW: looks like that package is only used in data browser, so could likely be removed from the workbench package.json. And don't bother spending too much time on this on my account, though I would request that if you want to go with this approach that you expand on the above comment to make it clear as to why there would be any linkage between cookies and local storage.